### ISSUE TITLE
Weird behaviour when using `ConfigurationExtension` and standard config files

### ISSUE THREAD

**Issue by isalvadori** [eclipse-osgi-technology/osgi-test#696]
We noticed a weird behaviour when using a combination of a config defined in a test through the `ConfigurationExtension` and some configurations defined in a standard configurator bundle.
It looks like before the `afterEach` method is called, also the configurations in the configurator bundle are deleted, resulting in exceptions, when trying, for instance, to drop a db in the `afterEach` method.

**Comment by kriegfrj** [comment]
@stbischof any ideas?

**Comment by stbischof** [comment]
There is a cleanup mechanism

**Comment by timothyjward** [comment]
Just to add, the presence of the `ConfigurationExtension` also causes instability in configurations provided by File Install, with those configurations being spuriously updated for each test. This can then impact the reliability of subsequent tests.

**Comment by stbischof** [comment]
One task of the `ConfigurationExtension` is to record the Configurations made in a test and reset them after each test.

https://github.com/osgi/osgi-test/blob/0e20db431c9857bd743c79fbc9f1ea4ce0211690/org.osgi.test.junit5.cm/src/main/java/org/osgi/test/junit5/cm/ConfigurationExtension.java#L318C15-L318C20

**Comment by timothyjward** [comment]
> One task of the ConfigurationExtension is to record the Configurations made in a test and reset them after each test

Agreed - but shouldn't this only apply to configurations injected using `@InjectConfiguration` or created using `@WithConfiguration`? If they're installed by a different management agent then I think the test framework should leave them alone.

**Comment by stbischof** [comment]
this would be a implementation that is much more what the user expect.

**Comment by stbischof** [comment]
should be fixed by https://github.com/osgi/osgi-test/pull/777

**Comment by juergen-albert** [comment]
Ilenia can retest in half a year :-(

**Comment by timothyjward** [comment]
The latest changes have definitely fixed this for me

**Comment by timothyjward** [comment]
Are we ok to close this issue now?
